src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -230,14 +230,37 @@ public async Task<bool> TryDownloadEmbeddedFile(string pluginKey, string embedde
             }, uri);
         }
 
-        public Task<AnalysisCacheMsg> DownloadCache(string projectKey, string branch)
+        public Task<IList<SensorCacheEntry>> DownloadCache(string projectKey, string branch)
         {
             _ = projectKey ?? throw new ArgumentNullException(nameof(projectKey));
             _ = branch ?? throw new ArgumentNullException(nameof(branch));
 
             logger.LogDebug(Resources.MSG_DownloadingCache, projectKey, branch);
             var uri = GetUri("api/analysis_cache/get?project={0}&branch={1}", projectKey, branch);
-            return downloader.DownloadStream(uri).ContinueWith(x => x.IsFaulted || x.Result == null ? null : AnalysisCacheMsg.Parser.ParseFrom(x.Result));
+            return downloader.DownloadStream(uri).ContinueWith(ParseCacheEntries);
+        }
+
+        private IList<SensorCacheEntry> ParseCacheEntries(Task<Stream> task)
+        {
+            if (task.IsFaulted || task.Result is not { } dataStream)
```

Comment:
Would the code be nicer if we would await the task instead of using .Result?

Reply:
I don't know. I think awaiting will throw an exception and this way we can just check for `IsFaulted`. I think it will be different, and not necessarily easier to read.